library/Translate/src/Juvix/Contextify/Passes.hs

```diff
@@ -13,20 +13,42 @@ import qualified Juvix.Sexp.Structure.Frontend as Structure
 import Juvix.Sexp.Structure.Lens
 import qualified StmContainers.Map as STM
 
+type ExpressionIO m = (Env.ErrS m, Env.HasClosure m, MonadIO m)
+
+type Expression m = (Env.ErrS m, Env.HasClosure m)
+
+
+--------------------------------------------------------------------------------
+-- Open Transformation
+--------------------------------------------------------------------------------
+
+-- The resolveModule pass, looks over any S-expression, searching for
+-- Atoms or open forms. In the case for Atoms, we see if there is any
+-- module that it should be qualified to.
+
+-- So for example if we have @sig foo : int -> int@ the @int@ atom
+-- really belongs to perhaps @Prelude.Michelson.int@. Thus we will do
+-- the conversion there. Further we change the @->@ atom into
+-- @Prelude.->@ for the same reason.
+
+-- In the open case we note the open into the closure, and then remove
+-- the open from the source code
+
+-- - BNF input form:
+--   1. (:open-in Foo body)
+--   2. unqualified-foo
+-- - BNF output form:
+--   1. body
+--   2. qualified-foo
+
+-- Note that the qualification of foo to Bar.foo if the symbol is from
+-- an open module
+
 resolveModule ::
```

Comment:
Not a question for this PR, but more long-term thought:  do we search by most-recently-opened modules first?  Or least-recently?  Or something else?  Would it make sense (if we don't already) to check whether there's only exactly one possible resolution, so that people can't accidentally intend to mean one thing but get another?

Reply:
If a user opens 2 overlapping modules, the code will error currently. I have logic in the context that deals with overlapping names. If a user introduces an ambiguity via interactive definition creation, the context will catch this and error if in some module it introduces an ambiguity.

With that said, the system is fine with `TopLevel` ambiguities, and `Opens` replacing local resolution, as those are puropsfully intended